File: backend/src/routes/reservas.ts
import { Router } from "express";
import { z } from "zod";

const router = Router();

// Placeholder en memoria - se reemplaza por Prisma/Postgres en el siguiente paso.
const reservas: Array<{ id: string; claseId: string; nombreCliente: string }> = [];

const crearReservaSchema = z.object({
  claseId: z.string().min(1),
  nombreCliente: z.string().min(2),
});

router.get("/", (_req, res) => {
  res.json(reservas);
});

router.post("/", (req, res) => {
  const parsed = crearReservaSchema.safeParse(req.body);
  if (!parsed.success) {
    return res.status(400).json({ error: parsed.error.flatten() });
  }

  const nueva = {
    id: String(reservas.length + 1),
    ...parsed.data,
  };
  reservas.push(nueva);
  res.status(201).json(nueva);
});

export default router;

File: backend/app/routers/reservas.py
from pydantic import BaseModel, Field

from fastapi import APIRouter

router = APIRouter(prefix="/api/reservas", tags=["reservas"])

# Placeholder en memoria - se reemplaza por la base de datos (SQLModel) en el
# siguiente paso.
reservas: list[dict] = []


class CrearReserva(BaseModel):
    clase_id: str = Field(min_length=1)
    nombre_cliente: str = Field(min_length=2)


@router.get("/")
def listar_reservas():
    return reservas


@router.post("/", status_code=201)
def crear_reserva(datos: CrearReserva):
    nueva = {"id": str(len(reservas) + 1), **datos.model_dump()}
    reservas.append(nueva)
    return nueva
